support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -224,16 +224,37 @@ function getAmountsTestParticipations(
 		return urlTest;
 	}
 
-	const { test } = settings.amounts?.[countryGroupId] ?? {};
+	if (!settings.amounts) {
+		return null;
+	}
+
+	const AMOUNTS = settings.amounts;
+
+	// Two-tier amounts - check for live country test, else get a region test
+	let targetTest = AMOUNTS.find((t) => t.target === country && t.isLive);
```

Comment:
So it looks like the model now has an array of tests. Question, can a country test be configured in the RRCP to target multiple countries? If so will that result in multiple items in the array of tests, one for each country the test is targeted towards? It appears so from this look up.

If this is the case a suggestion for a future change to the model would be for the test to have a single entry in the array of tests and for the `target` property to be an array of country codes.

So the lookup could be

```
let targetTest = AMOUNTS.find((t) => t.target.includes(country) && t.isLive);
```

I'm suggesting this as I'd be concerned that this amounts array could become pretty big if we start targeting multiple countries (eg. VAT countries), and we're shipping all of these tests to the browser in the HTML.

Reply:
`can a country test be configured in the RRCP to target multiple countries?` - no. An amounts test will target either a single region, or a single country.

`and we're shipping all of these tests to the browser in the HTML` - yes. They all end up in the window.guardian.settings.amounts object/array. I can't see a way around this, given that the Supporter+ landing page includes a region selector where users can change the currency they wish to donate in. I think there may be a way to restrict the data we push into window.guardian.settings.amounts to just the regions and the user's current country if an amounts test exists for that country?

Reply:
> `can a country test be configured in the RRCP to target multiple countries?` - no. An amounts test will target either a single region, or a single country.

Given a Country Test can only target one country then my original optimisation suggestion may be a moot point.

> `and we're shipping all of these tests to the browser in the HTML` - yes. They all end up in the window.guardian.settings.amounts object/array. I can't see a way around this, given that the Supporter+ landing page includes a region selector where users can change the currency they wish to donate in. I think there may be a way to restrict the data we push into window.guardian.settings.amounts to just the regions and the user's current country if an amounts test exists for that country?

I did think about whether we need to ship all tests to the browser, or whether they could be filtered on the server. Originally I suspected they'd all need be available on `window.guardian.settings.amounts` so we can handle when users come via amounts tests in the Banner/Epic, as I'd assumed the amounts test allocation on the checkout was based on the test/cohort present in the `acquisitionData` and we did a lookup using these values, however it looks like we just replicate the AB test allocation logic on the checkout and the banner/epic - so as long as they get the same input (amounts config and user MVT id) the outcome will be the same.

Reply:
I think we could do with assessing the impact of shipping a large list of amount tests to the browser and whether it can be optimised, I can see this list getting pretty big (particularly with the VAT proposal), one impact is it'll bloat the data in our cloudwatch logs significantly.

Reply:
Hi George. Question: how much of this problem will go away if/when we get into a world of 2-step checkout, where users coming from a Banner/Epic interaction will go directly to the step-2-checkout page (where we won't have to care about all the amounts tests) and the amounts tests only have to be supplied for users who go directly to the step-1-landing page?

I do like the idea of filtering tests on the server, but I don't know how to implement such a filter in Scala

`I think we could do with assessing the impact of shipping a large list of amount tests to the browser` - grateful for details on how I could perform this testing

Reply:
@KaliedaRik I'm not sure the 2 step checkout will help here, as looking at designs even if we drop a user on Page 2 there'll be the option to adjust their choice by stepping back to Page 1 where the amounts will need to be available. Of course. a solution here is to remove that ability to adjust the amounts for users landing directly on Page 2, but that's a whole discussion.

It'd be interesting to asses the impact of delivering something like the Contributions Only Country Targeted tests in the Amounts Test config, as there'll be around ~45 country targeted tests here. I was thinking we could test on CODE by mocking the `amounts-v2` JSON in S3 to have this volume of tests. We could then test this branch (using `amounts-v2`  vs main (using `amounts-v1`) using Chrome performance profiler?